### PR TITLE
Review property order

### PR DESCRIPTION
**Pull request by mbenson**
## Hey, I just made a Pull Request!

Reimplemented review content generation function to prefer property order from UI Schema.
Addresses https://github.com/backstage/backstage/issues/16062 

image

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [X] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder/src/components/MultistepJsonForm/ReviewStep.tsx [hunk 1]
```diff
@@ -25,38 +25,37 @@ export function getReviewData(
...
     }
-  }
+    if (value !== undefined) {
+      reviewData[key] = value;
+    }
+    formPropertyNames.delete(key);
+  };
+  uiSchemas
+    .filter(uiSchema => formPropertyNames.has(uiSchema.name))
+    .forEach(uiSchema => handleSchema(uiSchema.name, uiSchema));
```
Comment (benjdlambert) [inline plugins/scaffolder/src/components/MultistepJsonForm/ReviewStep.tsx:54]: Can we simplify both of these `.forEach`s into one `for` loop instead?

Think there's a little too much indirection and it makes the code quite hard to understand I think now.

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Hey :wave: thanks for this!

Just a quick high level comment about the structure. Feeling that we should be able to do this in one `for` loop rather than multiple `forEach`

File: plugins/scaffolder/src/components/MultistepJsonForm/ReviewStep.tsx [hunk 2]
```diff
@@ -25,36 +25,38 @@ export function getReviewData(
   uiSchemas: UiSchema[],
```
Comment (mbenson) [inline plugins/scaffolder/src/components/MultistepJsonForm/ReviewStep.tsx:25]: Wow, this diff is really unnecessarily noisy. A `git diff --ignore-all-space` shows that the only material changes in this iteration are like five lines in the beginning of the loop. The rest is whitespace from shifting indentation.

**Comment by mbenson** [comment]
Second approach preserves as much of original code as possible by just changing the source of the `for` loop.

**Review by benjdlambert — APPROVED** [review]
Nice thanks! :pray:

**Pull request merged**